web3/exceptions.py

```diff
@@ -2,7 +2,22 @@
 import time
 
 
-class BadFunctionCallOutput(Exception):
+class Web3Exception:
+    """
+    Exception mixin inherited by all exceptions of web3.py
+
+    This allows::
+
+        try:
+            some_call()
+        except Web3Exception as e:
+            # deal with web3 exception
+        except:
+            # deal with other exceptions
+    """
+
+
+class BadFunctionCallOutput(Exception, Web3Exception):
```

Comment:
I'd advocate for `Web3Exception` to inherit from `Exception` and to keep the inheritance chain for all of these child exception classes to only have a single parent of `Web3Exception`.

Reply:
Thanks for permission to break BC, notes added in new v6 migration document.